docs/architecture/distributed-cloud-native-apps-containers/api-gateways/gateway-patterns.md

```diff
@@ -0,0 +1,164 @@
+author: 
+ms.date: 04/25/2024
+---
+
+# Gateway patterns
+
+[!INCLUDE [download-alert](../includes/download-alert.md)]
+
+!TODO - sorry couldn't get these done in time. Should be easy to refresh from my powerpoint.
+
+In a cloud-native system, front-end clients (mobile, web, and desktop applications) require a communication channel to interact with independent back-end microservices.
+
+What are the options?
+
+To keep things simple, a front-end client could *directly communicate* with the back-end microservices.
+
+![Direct client to service communication](./media/direct-client-to-service-communication.png)
+
+**Figure 13-1**. Direct client to service communication
+
+With this approach, each microservice has a public endpoint that is accessible by front-end clients. In a production environment, you'd place a load balancer in front of the microservices, routing traffic proportionately.
+
+While simple to implement, direct client communication would be acceptable only for simple microservice applications. This pattern tightly couples front-end clients to core back-end services, opening the door for many problems, including:
+
+- Client susceptibility to back-end service refactoring.
+- A wider attack surface as core back-end services are directly exposed.
+- Duplication of cross-cutting concerns across each microservice.
+- Overly complex client code - clients must keep track of multiple endpoints and handle failures in a resilient way.
+
+Instead, a widely accepted cloud design pattern is to implement an [API Gateway Service](../microservices/architect-microservice-container-applications/direct-client-to-microservice-communication-versus-the-api-gateway-pattern.md) between the front-end applications and back-end services.
+
+![API Gateway Pattern](./media/api-gateway-pattern.png)
+
+**Figure 13-2**. API gateway pattern
+
+In the previous figure, note how the API Gateway service abstracts the back-end core microservices. Implemented as a web API, it acts as a *reverse proxy*, routing incoming traffic to the internal microservices.
+
+The gateway insulates the client from internal service partitioning and refactoring. If you change a back-end service, you accommodate for it in the gateway without breaking the client. It's also your first line of defense for cross-cutting concerns, such as identity, caching, resiliency, metering, and throttling. Many of these cross-cutting concerns can be off-loaded from the back-end core services to the gateway, simplifying the back-end services.
+
+Care must be taken to keep the API Gateway simple and fast. Typically, business logic is kept out of the gateway. A complex gateway risks becoming a bottleneck and eventually a monolith itself. Larger systems often expose multiple API Gateways segmented by client type (mobile, web, desktop) or back-end functionality. The [Backend for Frontends](/azure/architecture/patterns/backends-for-frontends) pattern provides direction for implementing multiple gateways.
```

Comment:
Same here Link not working for Backend for Frontends pattern.

Reply:
When it gets merged into the docs the learn.microsoft.com gets added to it, so it will point to https://learn.microsoft.com/en-us/azure/architecture/patterns/backends-for-frontends

Reply:
I see got it then I will try to append "learn.microsoft.com" on my browser  when checking links. thanks